File: CS/DiagramBindToList/Models/Edge.cs
using DevExpress.Web.ASPxDiagram;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace DiagramBindToList.Models
{
    public class Edge
    {
        public int ID { set; get; }
        public string Text { set; get; }
        public int? FromID { set; get; }
        public int? ToID { set; get; }
        //public ConnectorLineType LT { get; set; }

        public Edge(int id, string text, int fromId, int toId)
        {
            ID = id;
            Text = text;
            FromID = fromId;
            ToID = toId;
        }

        public Edge() { }
    }
}

File: VB/DiagramBindToListVB/Models/Edge.vb
Public Class Edge
    Public Property ID As Integer
    Public Property Text As String
    Public Property FromID As Integer?
    Public Property ToID As Integer?

    Public Sub New(ByVal id As Integer, ByVal text As String, ByVal fromId As Integer, ByVal toId As Integer)
        Me.ID = id
        Me.Text = text
        Me.FromID = fromId
        Me.ToID = toId
    End Sub

    Public Sub New()
    End Sub
End Class